moveit_core/constraint_samplers/CMakeLists.txt

```diff
@@ -28,17 +28,19 @@ if(CATKIN_ENABLE_TESTING)
   find_package(orocos_kdl REQUIRED)
   find_package(angles REQUIRED)
   find_package(tf2_kdl REQUIRED)
-
+  find_package(rostest REQUIRED)
   include_directories(SYSTEM ${orocos_kdl_INCLUDE_DIRS} ${angles_INCLUDE_DIRS} ${tf2_kdl_INCLUDE_DIRS})
 
-  catkin_add_gtest(test_constraint_samplers
+  add_rostest_gtest(test_constraint_samplers
+    test/constraint_samplers.test
```

Comment:
This seems like an unorthodox addition to `moveit_core` as I didn't see any instances of `rostest` in the package. I guess all `rostest` tests are categorized strictly differently than unit tests from the perspective of MoveIt conventions, but that's just a hunch regarding the status quo. 

If you think this is a serious violation, then I need to move the new added tests to a convenient place in `moveit_ros`.

Reply:
@v4hn, could you review as well and comment on this point? IMO, `moveit_core` isn't ROS-agnostic for a while already, e.g. using ROS logging in a few places. However, ROS parameters are actually not yet used in `moveit_core`.

Reply:
I fully agree. If we allow ros parameters here (and we seem to agree we want to do so) then we also need to allow tests for them, so both reading the param as well as adding the ros test are fine by me.

Maybe the ros2 sync will split it out into a separate framework structure, but for this repo I will just merge this as is now.